packages/@aws-cdk/aws-autoscaling/lib/auto-scaling-group.ts

```diff
@@ -1315,6 +1625,74 @@ export class AutoScalingGroup extends AutoScalingGroupBase implements
       metrics: group._metrics?.size !== 0 ? [...group._metrics].map(m => m.name) : undefined,
     }));
   }
+
+  private getLaunchSettings(launchConfig?: CfnLaunchConfiguration, launchTemplate?: ec2.ILaunchTemplate, mixedInstancesPolicy?: MixedInstancesPolicy)
```

Comment:
I am wondering if it is better to have a separate file for launch template and launch configuration because this file is getting bigger.

Reply:
I am not sure how practical it is since this design puts LC and LT under one single class, and I don't think we can break a class definition into multiple files like `partial class` in C#.